src/components/general/TextInput/index.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import styles from './styles.less';
+import classNames from 'classnames';
+import { ErrorIcon } from 'index';
+
+type TextInputProps = {
+    disabled?: boolean;
+    error?: boolean;
+    errorMessage?: string;
+    isOptional?: boolean;
+    placeholder?: string;
+    label?: string;
+    helperText?: string;
+    onChange: (newValue: string) => void;
+    value?: string;
+    icon?: any;
+    onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
+    onIconAction?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+};
+
+const TextInput = React.forwardRef<HTMLInputElement | null,React.PropsWithChildren<TextInputProps>>(
+    (
+        {
+            disabled = false,
+            error,
+            errorMessage,
+            isOptional = false,
+            placeholder = '',
+            label,
+            onChange,
+            value = '',
+            icon,
+            onBlur,
+            onIconAction,
+            helperText = '',
+            ...props
+        },
+        ref
+    ) => {
+        const [focus, setFocus] = React.useState(false);
+        const inputRef =
+            (ref as React.MutableRefObject<HTMLInputElement | null>) ||
+            React.useRef<HTMLInputElement | null>(null);
+
+        const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+            if (!disabled) {
+                const newValue = event.target.value;
+                onChange(newValue);
+            }
+        };
+
+        const setInputFocus = () => {
+            if (inputRef.current && !disabled) {
+                inputRef.current.focus();
+                setFocus(true);
+            }
+        };
+        const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+            if (!disabled) {
+                setFocus(false);
+                onBlur && onBlur(event);
+            }
+        };
+
+        const inputLabel = label && <label>{label}</label>;
+
+        const inputIcon = React.useMemo(() => {
+            if (!error && !icon) {
+                return null;
+            }
+            const inputIcon = error ? (
+                <ErrorIcon outline={false} color="#FF3B3B" cursor="default" />
+            ) : (
+                icon
+            );
+            return (
+                <div className={styles.icon} onClick={onIconAction}>
```

Comment:
Perhaps this could use the new IconButton component?

Reply:
The icon button is too big to be placed inside the text input component